homeassistant/components/light/vera.py

```diff
@@ -0,0 +1,181 @@
+                " was not found in config"
+            )
+            return False
+
+        device_data = config.get('device_data', None)
+
+        controller = veraApi.VeraController(base_url)
+        devices = controller.get_devices('Switch')
+
+        lights = []
+        for device in devices:
+            extra_data = get_extra_device_data(device_data, device.deviceId)
+            exclude = False
+            if extra_data:
+                exclude = extra_data.get('exclude', False)
+
+            if exclude is not True:
+                lights.append(VeraLight(device, extra_data))
+
+        add_devices_callback(lights)
+    # pylint: disable=broad-except
+    except Exception as inst:
+        _LOGGER.error("Could not find Vera lights: %s", inst)
+        return False
+
+
+def get_extra_device_data(device_data, device_id):
+    """ Gets the additional configuration data by Vera device Id """
+    if not device_data:
+        return None
+
+    for item in device_data:
+        if item.get('vera_id') == device_id:
+            return item
+
+    return None
+
+
+class VeraLight(ToggleDevice):
+    """ Represents a Vera light """
+    is_on_status = False
+    # for debouncing status check after command is sent
+    last_command_send = 0
+    extra_data = None
+
+    def __init__(self, vera_device, extra_data=None):
+        self.vera_device = vera_device
+        self.extra_data = extra_data
+
+    @property
+    def name(self):
+        """ Get the mame of the light. """
+        if self.extra_data and self.extra_data.get('name'):
+            return self.extra_data.get('name')
+        return self.vera_device.name
+
+    @property
+    def state_attributes(self):
+        attr = super().state_attributes
+
+        if self.vera_device.has_battery:
+            attr['Battery'] = self.vera_device.battery_level + '%'
+
+        if self.vera_device.is_armable:
+            armed = self.vera_device.refresh_value('Armed')
+            attr['Armed'] = 'True' if armed == '1' else 'False'
+
+        if self.vera_device.is_trippable:
+            last_tripped = self.vera_device.refresh_value('LastTrip')
+            trip_time_str = time.strftime(
+                "%Y-%m-%d %H:%M",
+                time.localtime(int(last_tripped))
+            )
+
+            attr['Last Tripped'] = trip_time_str
+
+            tripped = self.vera_device.refresh_value('Tripped')
+            attr['Tripped'] = 'True' if tripped == '1' else 'False'
+
+        attr['Vera Device Id'] = self.vera_device.vera_device_id
+
+        return attr
+
+    def turn_on(self, **kwargs):
+        self.last_command_send = time.time()
+        self.vera_device.switch_on()
+        self.is_on_status = True
+
+    def turn_off(self, **kwargs):
+        self.last_command_send = time.time()
+        self.vera_device.switch_off()
+        self.is_on_status = False
+
+    @property
+    def is_on(self):
+        """ True if device is on. """
+        self.update()
```

Comment:
Okay, had a better look at it now. You don't have to call `update` here because it will only be called before we update the state. See [update_ha_state in ToggleDevice](https://github.com/balloob/home-assistant/blob/master/homeassistant/helpers/device.py#L70). Now it will talk to Vera twice each time it updates.